src/test/java/com/ibm/watson/litelinks/test/LitelinksTests.java

```diff
@@ -1494,6 +1498,46 @@ public void service_subclass_test() throws Exception {
         }
     }
 
+    // Clone of DummyService with different service class name
+    public static class AliasServiceImpl implements DummyServiceAlias.Iface {
+        public AliasServiceImpl() {}
+        @Override public String method_one(String arg1, DummyStruct arg2, boolean arg3) throws TException {
+            return defaultImpl.method_one(arg1, arg2, arg3);
+        }
+        @Override public DummyStruct method_two(int arg1, String arg2, ByteBuffer arg3) throws TException {
+            return defaultImpl.method_two(arg1, arg2, arg3);
+        }
+    }
+
+    @Test
+    public void service_class_alias_test() throws Exception {
+        // service class alias config from env var LL_SERVICE_CLASS_ALIASES:
+        // "com.ibm.watson.litelinks.test.thrift.DummyServiceAlias=com.ibm.watson.litelinks.test.thrift.DummyService"
+
+        // Create clone of DummyService with different service class name
+        Service aliasService = LitelinksService.createService(AliasServiceImpl.class, ZK, null).startAsync();
+        Service svc = null;
+        try {
+            aliasService.awaitRunning();
+
+            // Create and test a client with aliased service class
+            DummyService.Iface aliasClient = ThriftClientBuilder.<DummyService.Iface>newBuilder(DummyService.Client.class)
+                    .withZookeeper(ZK).withServiceName(DummyServiceAlias.class.getName())
+                    .buildOnceAvailable(5000l);
+            basic_test(aliasClient);
+
+            // try to start server instance with same svc name but different svc class
+            svc = LitelinksService.createService(TestThriftServiceImpl.class,
```

Comment:
I guess I'm a bit unclear here, is this the bit that would fail before because zookeeper already knows about a running service with a different class name?

> with same svc name but different svc class

With the same service name as what, the running `aliasService`?


I'm just missing the context about how these fit together, but also just ignore me and merge if this test is doing the thing that it needs to

Reply:
Oops I think this is a good catch again, thanks!... the service name arg on the line below should be `DummyServiceAlias.class.getName()` rather than `DummyService.class.getName()`

> I guess I'm a bit unclear here, is this the bit that would fail before because zookeeper already knows about a running service with a different class name?

Basically yes.. same service name but the published class name for that service would be different and so the new instance would refuse to join it.